src/aws/ec2Instances.ts

```diff
@@ -0,0 +1,39 @@
+import {describeAsg} from './autoscaling'
+import {Instance} from './types'
+
+let AWS = require('aws-sdk');
+let awsEc2 = new AWS.EC2();
+
+type InstanceFilter = (instances: Instance[]) => Instance;
+
+export function getInstances(asgName: string): Promise<string[]> {
+    const autoScalingRequest = describeAsg(asgName);
+    return autoScalingRequest.then(
+        function (data) {
+            const instances = data.AutoScalingGroups[0].Instances;
+            return instances.map(instance => instance.InstanceId);
+        },
+        function (error) {
+            console.log(error, error.stack, error.statusCode);
+            return error;
+        }
+    )
+}
+
+export function getSpecificInstance(instanceIds: string[], instanceFilter: InstanceFilter): Promise<Instance> {
+    console.log(`Fetching details for: ${instanceIds}`);
+    const params = { InstanceIds: instanceIds };
+    const requestPromise = awsEc2.describeInstances(params).promise();
+    return requestPromise.then(
+        function(data) {
+            const instanceArrays = data.Reservations.map(instanceArrays => instanceArrays.Instances);
```

Comment:
Minor: shadowing of constant names - it might be clearer to use a different name inside the map?